Patch by Robert Scheck <robert@fedoraproject.org> for jwhois >= 4.0, which adds various new
IDN top-level-domains, updates some whois servers of ccTLDs and corrects the answer charset
setting for some whois servers, too.

--- jwhois-4.0/example/jwhois.conf			2011-05-22 20:36:19.000000000 +0200
+++ jwhois-4.0/example/jwhois.conf.conf_update2		2011-05-22 23:46:21.000000000 +0200
@@ -122,7 +122,6 @@
 	"\\.ac\\.ae$" = "whois.aeda.net.ae";
 	"\\.mil\\.ae$" = "whois.aeda.net.ae";
 	"\\.gov\\.ae$" = "whois.aeda.net.ae";
-	"\\.1\\.7\\.9\\.e164\\.arpa$" = "whois.aeda.net.ae";
 	"\\.aero$" = "whois.aero";
 	"\\.af$" = "whois.cocca.cx";
 	"\\.ag$" = "whois.nic.ag";
@@ -192,7 +191,7 @@
 	}
 	"\\.dm$" = "whois.nic.dm";
 	"\\.do$" = "whois.nic.do";
-	"\\.dz$" = "www.nic.dz";
+	"\\.dz$" = "whois.nic.dz";
 	"\\.ec$" = "www.nic.ec";
 	"\\.edu$" = "whois.educause.edu";
 	"\\.ee$" = "whois.eenet.ee";
@@ -243,7 +242,7 @@
 	"\\.kp$" = "whois.kcce.kp";
 	"\\.kg$" = "whois.domain.kg";
 	"\\.ki$" = "whois.cocca.cx";
-	"\\.kr$" = "whois.krnic.net";
+	"\\.kr$" = "whois.kr";
 	"\\.kz$" = "whois.nic.kz";
 	"\\.la$" = "whois.nic.la";
 	"\\.lb$" = "cgi.aub.edu.lb";
@@ -292,13 +291,14 @@
 	"\\.pm$" = "whois.nic.pm";
 	"\\.pr$" = "whois.nic.pr";
 	"\\.pro$" = "whois.registrypro.pro";
-	"\\.ps$" = "www.nic.ps";
+	"\\.ps$" = "whois.pnina.ps";
 	"\\.pt$" = "whois.dns.pt";
 	"\\.pw$" = "whois.nic.pw";
 	"\\.re$" = "whois.nic.re";
 	"\\.ro$" = "whois.rotld.ro";
+	"\\.rs$" = "whois.rnids.rs";
 	"\\.edu\.ru$" = "whois.informika.ru";
-	"\\.ru$" = "whois.ripn.net";
+	"\\.ru$" = "whois.tcinet.ru";
 	"\\.rw$" = "www.nic.rw";
 	"\\.sa$" = "saudinic.net.sa";
 	"\\.sb$" = "whois.nic.sb";
@@ -317,6 +317,7 @@
 	"\\.st$" = "whois.nic.st";
 	"\\.su$" = "whois.ripn.net";
 	"\\.sv$" = "www.svnet.org.sv";
+	"\\.sy$" = "whois.tld.sy";
 	"\\.tc$" = "whois.adamsnames.tc";
 	"\\.tel$" = "whois.nic.tel";
 	"\\.tf$" = "whois.afnic.fr";
@@ -335,7 +336,7 @@
 		whois-server = "tvwhois.verisign-grs.com";
 		query-format = "domain $*";
         }
-	"\\.tw$" = "whois.twnic.net";
+	"\\.tw$" = "whois.twnic.net.tw";
 	"\\.ua$" = "whois.com.ua";
 	"\\.ug$" = "whois.co.ug";
 	"\\.ac\\.uk$" = "whois.ja.net";
@@ -357,9 +358,25 @@
 	"\\.vu$" = "www.vunic.vu";
 	"\\.wf$" = "whois.nic.wf";
 	"\\.ws$" = "whois.worldsite.ws";
+	"\\.xn--3e0b707e$" = "whois.kr";
+	"\\.xn--90a3ac$" = "whois.rnids.rs";
+	"\\.xn--clchc0ea0b2g2a9gcd$" = "whois.sgnic.sg";
+	"\\.xn--fiqs8s$" = "cwhois.cnnic.cn";
+	"\\.xn--fiqz9s$" = "cwhois.cnnic.cn";
+	"\\.xn--fzc2c9e2c$" = "whois.nic.lk";
+	"\\.xn--j6w193g$" = "whois.hkirc.hk";
+	"\\.xn--kprw13d$" = "whois.twnic.net.tw";
+	"\\.xn--kpry57d$" = "whois.twnic.net.tw";
+	"\\.xn--lgbbat1ad8j$" = "whois.nic.dz";
 	"\\.xn--mgbaam7a8h$" = "whois.aeda.net.ae";
+	"\\.xn--mgberp4a5d4ar$" = "whois.nic.net.sa";
+	"\\.xn--o3cw4h$" = "whois.thnic.co.th";
+	"\\.xn--ogbpf8fl$" = "whois.tld.sy";
+	"\\.xn--p1ai$" = "whois.tcinet.ru";
+	"\\.xn--xkc2al3hye2a$" = "whois.nic.lk";
+	"\\.xn--yfro4i67o$" = "whois.sgnic.sg";
+	"\\.xn--ygbi2ammx$" = "whois.pnina.ps";
 	"\\.yt$" = "whois.nic.yt";
-	"\\.yu$" = "www.nic.yu";
 	"\\.ac\\.za$" = "whois.ac.za";
 	"\\.org\\.za$" = "rwhois.org.za 4321";
 	"\\.co\\.za$" = "whois.co.za";
@@ -389,6 +406,7 @@
 
 	"\\.9\\.4\\.e164\\.arpa$" = "whois.enum.denic.de";
 	"\\.1\\.6\\.e164\\.arpa$" = "whois-check.enum.com.au";
+	"\\.1\\.7\\.9\\.e164\\.arpa$" = "whois.aeda.net.ae";
 }
 
 #
@@ -1095,20 +1113,6 @@
 		form-element = "nombre_d";
 	}
 
-	"www\\.nic\\.dz" {
-		http = "true";
-		http-method = "POST";
-		http-action = "/index.php?option=com_content&task=view&id=37&Itemid=51";
-		form-element = "domain_name";
-	}
-
-	"www\\.nic\\.yu" {
-		http = "true";
-		http-method = "GET";
-		http-action = "/cgi-bin/checkavail.cgi";
-		form-element = "domain";
-	}
-
 	"whois\\.dotster\\.com" {
 		answer-charset = "UTF-8";
 	}
@@ -1127,13 +1131,6 @@
 		form-element = "domsrch";
 	}
 
-	"www\\.nic\\.ps" {
-		http = "true";
-		http-method = "GET";
-		http-action = "/whois/domain_whois.php";
-		form-element = "dname";
-	}
-
 	"www\\.svnet\\.org\\.sv" {
 		http = "true";
 		http-method = "POST";
@@ -1179,6 +1176,26 @@
 	"whois\\.1api\\.net" {
 		answer-charset = "UTF-8";
 	}
+
+	"whois\\.kr" {
+		answer-charset = "EUC-KR";
+	}
+
+	"whois\\.nic\\.or\\.kr" {
+		answer-charset = "EUC-KR";
+	}
+
+	"whois\\.nic\\.dz" {
+		answer-charset = "UTF-8";
+	}
+
+	"whois\\.nic\\.lk" {
+		answer-charset = "ISO-8859-11";
+	}
+
+	"whois\\.iana\\.org" {
+		answer-charset = "UTF-8";
+	}
 }
 
 #
